netmiko/base_connection.py

```
@@ -1634,6 +1649,8 @@ def send_command(
         if normalize:
             command_string = self.normalize_cmd(command_string)
 
+        log.info("In send_command, cmd:{}, read_timeout:{}".format(command_string, read_timeout))
```

related to below code, we should change _prompt_handler to raise exception if find_prompt raises exception. this is what we changed in netmiko2 as well

```
        if expect_string is not None:
            search_pattern = expect_string
        else:
            search_pattern = self._prompt_handler(auto_find_prompt)

---

in netmiko2 we added a log.info statement when the exception was swallowed 
https://github.com/cafykit/netmiko/blob/7fcc795484e4ccdecc7b3e8b6e365b1cc5c7afb7/netmiko/base_connection.py#L1095

---

added the similar log.info line in netmiko4

---

I think it is suggested to pick up the changes from cxr_connection.py.
[netmiko/cisco/cisco_vxr_ssh.py](https://github.com/cafykit/netmiko-v4/pull/14/files#diff-c134a175762ced46d28ff2ae0a5f08fc4ef2669c14543bff8e6da37d4cb3a72b) 

the reason is this code is absolutely wrong in base file. I understand that we want to minimize the changes form original netmiko but this is not the way underlaying layers behave..

netmiko -> paramiko -> router.

if the channel is alive and def _send doesnt fail at paramiko that means anything sent will arrive and we must get a pattern.
if router is stuck it will not respond to 1st or nth. 
so , if we send 1 "\n" or a null byte ,  and we wait for response untill session is alive.

---

created seperate PR for find_prompt() changes to amke it independent https://github.com/cafykit/netmiko-v4/pull/15